cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactInspector.java

```diff
@@ -321,6 +409,29 @@ private String getPluginName(Class<?> cls) {
   }
 
   /**
+   * Extracts and returns set of endpoints in the plugin.
+   * @throws BadRequestException if there are duplicate endpoints found.
+   */
+  private Set<String> getPluginEndpoints(Class<?> cls) throws BadRequestException {
+    Set<String> endpoints = new HashSet<>();
+    Method[] methods = cls.getDeclaredMethods();
+    for (Method method : methods) {
+      Path pathAnnotation = method.getAnnotation(Path.class);
+      // method should have path annotation else continue
+      if (pathAnnotation == null) {
+        continue;
+      } else {
+        if (!endpoints.add(pathAnnotation.value())) {
```

Comment:
yes but you should also get the method arguments here and validate so that it fails and the artifact isn't even deployed. Otherwise it will fail much later at runtime which is bad.